### PR TITLE
[WIP] Adding an option to remove seconds hand in clock activity

### PR DESCRIPTION
**Pull request by ricknjacky**
I'm taking a step by step process approach to this issue ( #885 )

1. I planned on adding the button, svg for the button, .ini file changes, Which I have done already
2. Adding the functionality, that will eventually hide the seconds hand 

will start the 2nd step after changes in 1st step are approved, waiting for feedback

### CODE REVIEW

File: activities/Clock.activity/icons/second-hand.svg [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<svg id="SvgjsSvg1001" width="288" height="288" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs"><defs id="SvgjsDefs1002"></defs><g id="SvgjsG1008" transform="matrix(0.9167,0,0,0.9167,11.99519256754698,11.995237874771647)"><svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 311.661 311.661" viewBox="0 0 311.661 311.661" width="288" height="288"><path d="M155.661,65.431c-50,0-90.8,40.8-90.8,90.8s40.8,90.4,90.8,90.4s90.4-40.8,90.8-90.8
```
Comment (llaske) [inline activities/Clock.activity/icons/second-hand.svg:1]: I'm not sure the gears is pertinent here.
I suggest to use two icons for this new button: one with three hands when the seconds hand is here, one with two hands when the seconds in not here.
Ensure that the icon is free/libre license or is derivative work on a free/libre icon.

File: activities/Clock.activity/locale.ini [hunk 2]
```diff
@@ -45,7 +45,8 @@ TutoSetTimeTitle=Set-Time Mode
 TutoSetTimeContent=Click this icon to switch to set-time mode. In this mode you could move the hands to display the hour you want
 TutoSetTimeGameTitle=Game Mode
 TutoSetTimeGameContent=Click this icon to switch to set-time-Game mode. In this mode, the time is displayed in number format and you have to move the hands to set it in the matching hour
-
+TutoSetSecondsTitle = Seconds Hand Hide.
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:48]: Not dot at the end of title.

File: activities/Clock.activity/locale.ini [hunk 3]
```diff
@@ -213,6 +216,8 @@ TutoSetTimeTitle=Changer l'heure
 TutoSetTimeContent=Cliquez sur cette icône pour passer en mode changement d'heure. Dans ce mode vous pouvez déplacer les aiguilles pour indiquer l'heure que vous souhaitez
 TutoSetTimeGameTitle=Devinez l'heure
 TutoSetTimeGameContent=Cliquez sur cette icône pour passer au mode jeu. Dans ce mode, une heure est affichée au format numérique et vous devez déplacer les aiguilles pour les positionner sur l'heure correspondante
+TutoSetSecondsTitle = Peau de seconde main.
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:219]: Except if you're native from the target language, it's better to use the English localization everywhere. i.e. avoid automatic translation.

File: activities/Clock.activity/locale.ini [hunk 4]
```diff
@@ -213,6 +216,8 @@ TutoSetTimeTitle=Changer l'heure
...
+TutoSetSecondsContent= Cliquez sur cette icône pour masquer la trotteuse.
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:220]: Same remark on localization: avoid using automatic translation.

File: activities/Clock.activity/locale.ini [hunk 5]
```diff
@@ -260,7 +265,8 @@ TutoSetTimeTitle=modo de hora definida
 TutoSetTimeContent=Kliknij tę ikonę, aby przejść do trybu ustawiania czasu
 TutoSetTimeGameTitle=Tryb ustawiania czasu gry
 TutoSetTimeGameContent=Kliknij tę ikonę, aby przejść do trybu gry ustawionego czasu
-
+TutoSetSecondsTitle = Skóra z drugiej ręki.
+TutoSetSecondsContent= Kliknij tę ikonę, aby ukryć drugie rozdanie.
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:269]: Same remark on localization: avoid using automatic translation.

File: activities/Clock.activity/locale.ini [hunk 6]
```diff
@@ -307,3 +313,5 @@ TutoSetTimeTitle=definir modo de hora
 TutoSetTimeContent=Clique neste ícone para alternar para o modo de tempo definido
 TutoSetTimeGameTitle=Jogo de hora marcada
 TutoSetTimeGameContent=Clique neste ícone para alternar para o modo de hora definida
+TutoSetSecondsTitle = Couro de segunda mão.
+TutoSetSecondsContent= Clique neste ícone para ocultar o ponteiro dos segundos.
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:317]: Same remark on localization: avoid using automatic translation.

**Review by llaske — CHANGES_REQUESTED** [review]
Thanks for this first step. Add [WIP] in the title of your PR to make clear that it's a work in progress.

File: activities/Clock.activity/locale.ini (hunk 3 above)
Comment (ricknjacky) [reply activities/Clock.activity/locale.ini:219]: So I type in the exact sentences in different languages as well? like the same which I wrote in English I should write in French ?

File: activities/Clock.activity/locale.ini (hunk 2 above)
Comment (ricknjacky) [reply activities/Clock.activity/locale.ini:48]: will do

File: activities/Clock.activity/icons/second-hand.svg (hunk 1 above)
Comment (ricknjacky) [reply activities/Clock.activity/icons/second-hand.svg:1]: wouldn't it be better if we just add one button and then the message i.e the content it shows on hover is different if the seconds hand is hidden and vice versa? 
just a suggestion
will go ahead with what you think is correct

Comment (llaske) [reply activities/Clock.activity/icons/second-hand.svg:1]: May be I was not clear: yes the idea is to have one button with two different possible background icon (one with two hands, one with three hands). Regarding "on hover" event it don't works on touch devices so we can't really rely on it.

File: activities/Clock.activity/locale.ini (hunk 3 above)
Comment (llaske) [reply activities/Clock.activity/locale.ini]: Yes. There is a localization platform on https://translate.sugarizer.org where native speakers could translate strings without any development skill.

Comment (ricknjacky) [reply activities/Clock.activity/locale.ini]: I have done the changes you requested please review :)

File: activities/Clock.activity/icons/second-hand.svg (hunk 1 above)
Comment (ricknjacky) [reply activities/Clock.activity/icons/second-hand.svg:1]: yes I agree, will do the changes soon

File: activities/Clock.activity/locale.ini [hunk 7]
```diff
@@ -45,8 +45,8 @@ TutoSetTimeTitle=Set-Time Mode
 TutoSetTimeContent=Click this icon to switch to set-time mode. In this mode you could move the hands to display the hour you want
 TutoSetTimeGameTitle=Game Mode
 TutoSetTimeGameContent=Click this icon to switch to set-time-Game mode. In this mode, the time is displayed in number format and you have to move the hands to set it in the matching hour
-TutoSetSecondsTitle = Seconds Hand Hide.
-TutoSetSecondsContent= Click this icon to hide the second hand.
+TutoSetSecondsTitle = Seconds Hand Hide
+TutoSetSecondsContent= Click this icon to hide the second hand
```
Comment (llaske) [inline activities/Clock.activity/locale.ini:49]: The button will be use not only to hide the second hand but also to show it again when it's hide.

**Comment by ricknjacky** [comment]
@llaske I have added the two images, they perform the switch between them but just once. However,I have given it a try. I request you to guide me on how to achieve the desired output i.e toggle effect of background image in here. And suggest changes if any.

File: activities/Clock.activity/index.html [hunk 8]
```diff
@@ -17,7 +17,16 @@
 <script>if (window.module) module = window.module;</script>
 <script data-main="js/loader" src="lib/require.js"></script>
 </head>
-
+<script>
```
Comment (llaske) [inline activities/Clock.activity/index.html:20]: No JavaScript should be included in the HTML files, it's not supported on some platforms (Chrome OS for example).
You should add an event listener like it's done for other events.

File: activities/Clock.activity/index.html [hunk 9]
```diff
@@ -17,7 +17,16 @@
...
+  function change() {
+    var abc = document.getElementById("seconds-button-before")
+    if (abc) {
+        abc.id = "seconds-button-after"
```
Comment (llaske) [inline activities/Clock.activity/index.html:24]: Not a good idea to change the id. You could use two different classes and switch from one to another.

File: activities/Clock.activity/icons/after.svg [hunk 10]
```diff
@@ -0,0 +1,7 @@
+<svg id="SvgjsSvg1001" width="288" height="288" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs"><defs id="SvgjsDefs1002"></defs><g id="SvgjsG1008" transform="matrix(0.9167,0,0,0.9167,11.995200000000011,11.995200000000011)"><svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 470 470" viewBox="0 0 470 470" width="288" height="288"><path d="M401.17,68.83C356.785,24.444,297.771,0,235,0S113.215,24.444,68.83,68.83C24.444,113.216,0,172.229,0,235
```
Comment (llaske) [inline activities/Clock.activity/icons/after.svg:1]: Icons must not be colored. You should use a more explicit name for the file.

File: activities/Clock.activity/icons/before.svg [hunk 11]
```diff
@@ -0,0 +1,7 @@
+<svg id="SvgjsSvg1001" width="288" height="288" xmlns="http://www.w3.org/2000/svg" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:svgjs="http://svgjs.com/svgjs"><defs id="SvgjsDefs1002"></defs><g id="SvgjsG1008" transform="matrix(0.9167,0,0,0.9167,11.995172036743156,11.995200000000011)"><svg xmlns="http://www.w3.org/2000/svg" enable-background="new 0 0 445.462 445.462" viewBox="0 0 445.462 445.462" width="288" height="288"><path d="M380.225,65.236C338.157,23.168,282.225,0,222.731,0S107.305,23.168,65.237,65.236S0,163.237,0,222.731
```
Comment (llaske) [inline activities/Clock.activity/icons/before.svg:1]: Icons must not be colored. You should use a more explicit name for the file.
Both icons should be identical except for the seconds hand.

File: activities/Clock.activity/index.html [hunk 12]
```diff
@@ -28,7 +37,8 @@
     <button class="toolbutton" id="write-date-button" title="Write date"></button>
 		<hr/>
 		<button class="toolbutton" id="set-time-button" title="Set Time"></button>
-		<button class="toolbutton" id="set-timeGame-button" title="Set Time to target"></button>
+    <button class="toolbutton" id="set-timeGame-button" title="Set Time to target"></button>
+    <button class="toolbutton" id="seconds-button-before" title="Hide Seconds Hand" onclick="change()"></button>
```
Comment (llaske) [inline activities/Clock.activity/index.html:41]: Icon should be localized to the right of the display date icon.

**Review by llaske — CHANGES_REQUESTED** [review]
See my comments.

**Comment by ricknjacky** [comment]
@llaske all the changes requested have been made

- The svg files are free/libre to use
-The onclick toggle works completely fine now

please review the changes, so that I can move to step 2

**Review by llaske — CHANGES_REQUESTED** [review]
Both icons should be identical except for the seconds hand.
Icon should be localized to the right of the display date icon.

**Comment by llaske** [comment]
Both icons should be identical except for the seconds hand.
Icon should be localized before the vertical bar.

**Comment by ricknjacky** [comment]
>Icon should be localized before the vertical bar.

Can you please tell me what exactly, i have to do in here?(please pardon my repeated queries)

**Comment by llaske** [comment]
Regarding the icon, you should find one with the seconds hand and use a tool like Inkscape to create a copy of the icon without the seconds hand.
For the location of this new icon, you should put it before the `hr` tag in the `index.html` file.

**Comment by llaske** [comment]
Fixed in https://github.com/llaske/sugarizer/commit/0aa669ed630a83ef629b0b3bc6b36107d19edad8

**Pull request closed without merging**